docs: spelling mistakes in the migration guide

puzzle-it-nu opened the pull request:
<!---
☝️ PR title should follow conventional commits (https://conventionalcommits.org)
-->

### 🔗 Linked issue
#2907 
### ❓ Type of change
Fix spelling mistakes

<!-- What types of changes does your code introduce? Put an `x` in all the boxes that apply. -->

- [x] 📖 Documentation (updates to the documentation or readme)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description
Fixes what I think are some spelling mistakes.

<!-- Describe your changes in detail -->
<!-- Why is this change required? What problem does it solve? -->
<!-- If it resolves an open issue, please link to the issue here. For example "Resolves #1337" -->

### 📝 Checklist

<!-- Put an `x` in all the boxes that apply. -->
<!-- If your change requires a documentation PR, please link it appropriately -->
<!-- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->

- [x] I have linked an issue or discussion.
- [ ] I have updated the documentation accordingly.

farnabaz commented:
Thanks for the PR, 
I'll wait for you to move it out of WIP

puzzle-it-nu commented:
Question:
There are numbered folders in the `docs/content/docs` folder, from 1 to 7 but the number '3' occurs twice.
Is that correct?

puzzle-it-nu commented:
@farnabaz I've made a mistake, commit 794379b should be reverted.
Do I make a revert commit or can I rebase and drop that commit? What is the preferred way todo this.

puzzle-it-nu commented:
Things I came across but don't know what to do with:
- frontmatter vs front-matter vs Front Matter
- pascal casing in 'title' frontmatter isn't consistent
- index.md 'write content in md, yaml, json' vs 'md yaml, json, csv' in the example

Spelling in the browser :smile:  says frontmatter should either be `front matter` or `front-matter`

puzzle-it-nu commented:
> Nice work @puzzle-it-nu 👍
> 
Thanks.

> > There are numbered folders in the docs/content/docs folder, from 1 to 7 but the number '3' occurs twice.
> 
> It is incorrect, and need to be fixed

Should the second '3' become '4' and so on?

> 
> > Do I make a revert commit or can I rebase and drop that commit? What is the preferred way todo this.
> 
> There isn't any preferred way, do as you prefer :)
> 

Ok, I'm in doubt whether it is really a problem. It looks more a visual indentation than that is has a function.
If the latter is the case then it can stay.

> > frontmatter vs front-matter vs Front Matter
> 
> I think we should use frontmatter all the time, as a single word since it refers to the metadata block

Ok, I'll change that where needed.

> 
> > index.md 'write content in md, yaml, json' vs 'md yaml, json, csv' in the example
> 
> Shall we add csv to index.md?

I'm in favour of that, will add it.

puzzle-it-nu commented:
@farnabaz two more questions/'problems'
- point 5 'Deploy' has 3 items but they are not numbered. All the other chapters do have numbered items. My suggestion would be to number them.
- there are a dozen or so files with a frontmatter meta that contains 2 times a `title:`, looking at the content3 site it looks like the second one wins. Problem is I don't know whether the first or second one is the correct one. For example: 5.Deploy/server.md

puzzle-it-nu commented:
@farnabaz about the titles makes sense, thanks

farnabaz commented on docs/content/docs/2.collections/1.collections.md:
```diff
@@ -1,7 +1,7 @@
 title: Define Content Collections
 navigation:
-  title: Define
+title: Define
```
```suggestion
  title: Define
```

puzzle-it-nu replied:
Suggestion added

farnabaz commented on docs/content/docs/2.collections/2.types.md:
```diff
@@ -1,7 +1,7 @@
 title: Collection Types
 navigation:
-  title: Types
+title: Types
```
```suggestion
  title: Types
```

puzzle-it-nu replied:
Suggestion added

farnabaz commented on docs/content/docs/2.collections/3.sources.md:
```diff
@@ -1,7 +1,7 @@
 title: Collection Sources
 navigation:
-  title: Sources
+title: Sources
```
```suggestion
  title: Sources
```

navigation title should be indented

puzzle-it-nu replied:
Suggestion added

farnabaz commented on docs/content/docs/3.files/2.yaml.md:
````diff
@@ -49,6 +49,7 @@ Now we can query authors:
 ```ts
 // Find a single author
 const theAuthor = await queryCollection('authors')
+// TODO: should `stem` be `name`?
````
`stem` is the file name without extension. we can keep stem

farnabaz commented on docs/content/docs/3.files/3.json.md:
````diff
@@ -57,6 +57,7 @@ Now we can query authors:
 ```ts
 // Find a single author
 const theAuthor = await queryCollection('authors')
+// TODO: should `stem` be `name`?
````
`stem` is the file name without extension. we can keep stem

puzzle-it-nu commented:
@farnabaz added your changes and also added numbering and fixed the double '3'

There are still a few spots where the wording feels odd to me but I'm not a native english speaker so that could just me.

farnabaz reviewed (approved):
LGTM 👍 
Thanks
